### PR TITLE
Create Reward Constants endpoint

### LINKED ISSUE #1356: Create Reward Constants endpoint
### Description

Create the following endpoint.

```
HTTP GET /api/v3/reward/constants
RPC  get.reward.constants
```

### Acceptance Criteria

- The endpoints listed above are created
- The response is matching as described below
- All the methods, variables, files are renamed appropriately
- Swagger is up-to-date
- Integration tests are up-to-date

### Additional information
*Request params:*
Not Applicable

*Responses:*

200 OK
```jsonc
{
    "data": {
        "rewardTokenID": "0000000000000000",
    },
    "meta": {},
}
```

400 Bad Request
```jsonc
{
  "error": true,
  "message": "Unknown input parameter(s): <param_name>"
}
```

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1356 

### How was it solved?
- [x] The endpoints listed above are created
- [x] The response is matching as described below
- [x] All the methods, variables, files are renamed appropriately
- [x] Swagger is up-to-date
- [x] Integration tests are up-to-date

### How was it tested?
- [x] Local

### CODE REVIEW

File: services/gateway/apis/http-version3/swagger/parameters/dynamicReward.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "blockHeight": {
```
Comment (sameersubudhi) [inline services/gateway/apis/http-version3/swagger/parameters/dynamicReward.json:2]:
```suggestion
    "rewardBlockHeight": {
```

File: services/gateway/shared/utils.js [hunk 2]
```diff
@@ -68,6 +68,8 @@ const transformParams = (type, params) => {
 			};
 		} else if (type === 'PoS') {
 			if (paramKey === 'name') value = { $ref: '#/parameters/validatorName' };
+		} else if (type === 'reward' && paramKey === 'height') {
+			value = { $ref: '#/parameters/blockHeight' };
```
Comment (sameersubudhi) [inline services/gateway/shared/utils.js:72]:
```suggestion
			value = { $ref: '#/parameters/rewardBlockHeight' };
```

File: services/blockchain-indexer/shared/dataService/business/dynamicReward/constants.js [hunk 3]
```diff
@@ -22,6 +22,20 @@ const getRewardTokenID = async () => {
...
 
+const getRewardConstants = async () => {
+	const response = {
+		data: {},
+		meta: {},
+	};
+
+	response.data = {
+		rewardTokenID: await getRewardTokenID(),
+	};
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/dataService/business/dynamicReward/constants.js:33]:
```suggestion
	response.data.rewardTokenID = await getRewardTokenID();
```

File: services/blockchain-connector/shared/sdk/dynamicReward.js [hunk 4]
```diff
@@ -36,6 +40,22 @@ const getRewardTokenID = async () => {
 	return rewardTokenID;
 };
 
+const getDefaultRewardAtHeight = async height => {
+	try {
+		// TODO: Update endpoint once exposed by sdk
```
Comment (sameersubudhi) [inline services/blockchain-connector/shared/sdk/dynamicReward.js:45]: Do we still need the TODO? 🤔 
```suggestion
```

File: services/blockchain-connector/shared/sdk/dynamicReward.js [hunk 5]
```diff
@@ -36,6 +40,22 @@ const getRewardTokenID = async () => {
...
+		const defaultReward = await invokeEndpoint('dynamicReward_getDefaultRewardAtHeight', { height });
```
Comment (sameersubudhi) [inline services/blockchain-connector/shared/sdk/dynamicReward.js:46]: This is also missing the error handling to switch to `reward_*****` endpoint.

File: services/blockchain-indexer/shared/dataService/business/dynamicReward/defaultReward.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+const getDefaultRewardAtHeight = async (params) => {
+	const response = {
+		data: {},
+		meta: {},
+	};
+	const defaultReward = await requestConnector('getDefaultRewardAtHeight', { height: params.height });
+	const rewardTokenID = await getRewardTokenID();
+
+	response.data = {
+		defaultReward: defaultReward.error ? null : defaultReward,
```
Comment (sameersubudhi) [inline services/blockchain-indexer/shared/dataService/business/dynamicReward/defaultReward.js:29]: Was this not supposed to be like this?
```
		defaultReward: defaultReward.reward,
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: services/gateway/apis/http-version3/swagger/parameters/dynamicReward.json (hunk 1 above)
Comment (priojeetpriyom) [reply services/gateway/apis/http-version3/swagger/parameters/dynamicReward.json:2]: The param is not dependant on reward module and can reused whenever we need to query for specific block height.

File: services/gateway/shared/utils.js (hunk 2 above)
Comment (priojeetpriyom) [reply services/gateway/shared/utils.js:72]: https://github.com/LiskHQ/lisk-service/pull/1377#discussion_r1043416179

File: services/gateway/apis/http-version3/swagger/parameters/dynamicReward.json (hunk 1 above)
Comment (sameersubudhi) [reply services/gateway/apis/http-version3/swagger/parameters/dynamicReward.json:2]: Then this shouldn't be defined within the param file specific to `dynamicReward`. Please move it to the appropriate file.

File: services/blockchain-indexer/shared/dataService/business/dynamicReward/defaultReward.js (hunk 6 above)
Comment (priojeetpriyom) [reply services/blockchain-indexer/shared/dataService/business/dynamicReward/defaultReward.js]: Sorry, all changes were not brought to this branch. Hence the confusion, updated base branch

File: services/blockchain-connector/shared/sdk/dynamicReward.js (hunk 5 above)
Comment (priojeetpriyom) [reply services/blockchain-connector/shared/sdk/dynamicReward.js]: https://github.com/LiskHQ/lisk-service/pull/1377#discussion_r1043418768

File: services/blockchain-connector/shared/sdk/dynamicReward.js (hunk 4 above)
Comment (priojeetpriyom) [reply services/blockchain-connector/shared/sdk/dynamicReward.js:48]: Better to keep it as the endpoint is still not running.

File: services/gateway/apis/http-version3/swagger/parameters/dynamicReward.json (hunk 1 above)
Comment (priojeetpriyom) [reply services/gateway/apis/http-version3/swagger/parameters/dynamicReward.json]: moved to `common.json`

**Review by sameersubudhi — APPROVED** [review]

**Review by nagdahimanshu — APPROVED** [review]
Looks good 👍

**Pull request merged**